query/query.go

```diff
@@ -31,27 +31,37 @@ type Q interface {
 	String() string
 }
 
+type SearchScope uint8
+
+const (
+	ScopeNone SearchScope = iota
+	ScopeFileName
+	ScopeFileContent
+	ScopeSymbol
+)
```

Comment:
one often uses https://godoc.org/golang.org/x/tools/cmd/stringer for these sort of enums.

Whenever we check for ScopeNone, it seems to be expanded into an OR of ScopeFileName and ScopeFileContent. I wonder if there is a nice way to represent this. Additionally, I wonder if we should include scopesymbol in that case. However, in those cases we may have duplicate results since the ranking part already takes int oaccount overlaps on symbols.